src/actions/index.js

```diff
@@ -34,17 +34,26 @@ export const LOAD_DEPENDENCY_INFO_FROM_DISK = 'LOAD_DEPENDENCY_INFO_FROM_DISK';
 export const ADD_DEPENDENCY_START = 'ADD_DEPENDENCY_START';
 export const ADD_DEPENDENCY_ERROR = 'ADD_DEPENDENCY_ERROR';
 export const ADD_DEPENDENCY_FINISH = 'ADD_DEPENDENCY_FINISH';
+export const ADD_DEPENDENCIES_START = 'ADD_DEPENDENCIES_START';
+export const ADD_DEPENDENCIES_ERROR = 'ADD_DEPENDENCIES_ERROR';
+export const ADD_DEPENDENCIES_FINISH = 'ADD_DEPENDENCIES_FINISH';
 export const UPDATE_DEPENDENCY_START = 'UPDATE_DEPENDENCY_START';
 export const UPDATE_DEPENDENCY_ERROR = 'UPDATE_DEPENDENCY_ERROR';
 export const UPDATE_DEPENDENCY_FINISH = 'UPDATE_DEPENDENCY_FINISH';
 export const DELETE_DEPENDENCY_START = 'DELETE_DEPENDENCY_START';
 export const DELETE_DEPENDENCY_ERROR = 'DELETE_DEPENDENCY_ERROR';
 export const DELETE_DEPENDENCY_FINISH = 'DELETE_DEPENDENCY_FINISH';
+export const DELETE_DEPENDENCIES_START = 'DELETE_DEPENDENCIES_START';
+export const DELETE_DEPENDENCIES_ERROR = 'DELETE_DEPENDENCIES_ERROR';
+export const DELETE_DEPENDENCIES_FINISH = 'DELETE_DEPENDENCIES_FINISH';
 export const SHOW_IMPORT_EXISTING_PROJECT_PROMPT =
   'SHOW_IMPORT_EXISTING_PROJECT_PROMPT';
 export const IMPORT_EXISTING_PROJECT_START = 'IMPORT_EXISTING_PROJECT_START';
 export const IMPORT_EXISTING_PROJECT_ERROR = 'IMPORT_EXISTING_PROJECT_ERROR';
 export const IMPORT_EXISTING_PROJECT_FINISH = 'IMPORT_EXISTING_PROJECT_FINISH';
+export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
+export const UPDATE_NOTIFICATION = 'UPDATE_NOTIFICATION';
+export const DELETE_NOTIFICATION = 'DELETE_NOTIFICATION';
```

Comment:
Ah, so I try to avoid actions like this; it's not always possible, but as I was saying above, I think Redux is nicest when the actions describe the sequence of things that happened. Notifications aren't a thing, they're a side-effect of another thing. So I think the notifications reducer should just be listening for actions that would trigger a notification (like `DELETE_DEPENDENCIES_FINISH` would update the notification reducer by removing that "deleting..." notification).

Reply:
Ok cool, I wasn't sure if it would make sense to have loads of unrelated actions aggregated in a single reducer if more functionality was added down the road that also required notifications. In the scope of the entire project though I think you're right, it's more intuitive to have the notifications reducer handle all notification related actions and not have to remember to manually dispatch them.